includes/class-ast-staging.php

```diff
@@ -0,0 +1,215 @@
+
+	/**
+	 * Handle site change notice actions
+	 *
+	 * @since 1.2.0
+	 */
+	public static function handle_site_change_notice_actions() {
+		if ( self::is_duplicate_site() && current_user_can( 'manage_options' ) ) {
+
+			if (
+				! empty( $_GET['_astnonce'] ) &&
+				wp_verify_nonce( sanitize_text_field( wp_unslash( $_GET['_astnonce'] ) ), 'ast_duplicate_site' ) &&
+				isset( $_GET['autoshare_duplicate_site'] )
+			) {
+				$duplicate_site = sanitize_text_field( wp_unslash( $_GET['autoshare_duplicate_site'] ) );
+				if ( 'update' === $duplicate_site ) {
+					self::set_autoshare_site_url_lock();
+				} elseif ( 'ignore' === $duplicate_site ) {
+					update_option( 'autoshare_ignore_duplicate_siteurl_notice', self::get_autoshare_site_url_lock_key() );
+				}
+				wp_safe_redirect( remove_query_arg( array( 'autoshare_duplicate_site', '_astnonce' ) ) );
+			}
+		}
+	}
+
+	/**
+	 * Displays a notice when plugin is being run on a different site, like a staging or testing site.
+	 *
+	 * @since 1.2.0
+	 */
+	public static function handle_site_change_notice() {
+		if (
+			self::is_duplicate_site() &&
+			current_user_can( 'manage_options' ) &&
+			self::get_autoshare_site_url_lock_key() !== get_option( 'autoshare_ignore_duplicate_siteurl_notice' )
+		) {
+			$ignore_url = wp_nonce_url( add_query_arg( 'autoshare_duplicate_site', 'ignore' ), 'ast_duplicate_site', '_astnonce' );
+			$update_url = wp_nonce_url( add_query_arg( 'autoshare_duplicate_site', 'update' ), 'ast_duplicate_site', '_astnonce' );
+			?>
+			<div class="notice notice-error">
+				<p>
+					<?php
+					printf(
+						// translators: 1$-2$: opening and closing <strong> tags. 3$-4$: Opening and closing link to production URL. 5$: Production URL.
+						esc_html__( 'It looks like this site has moved or is a duplicate site. %1$sAutoshare for Twitter%2$s has disabled publish tweets on this site to prevent tweets from a staging or test environment. %1$sAutoshare for Twitter%2$s considers %3$s%5$s%4$s to be the site\'s URL. ', 'autoshare-for-twitter' ),
+						'<strong>',
+						'</strong>',
+						'<a href="' . esc_url( self::get_site_url_from_source( 'autoshare' ) ) . '" target="_blank">',
+						'</a>',
+						esc_url( self::get_site_url_from_source( 'autoshare' ) )
+					)
+					?>
+				</p>
+				<p>
+					<a class="button button-primary" href="<?php echo esc_url( $ignore_url ); ?>">
+						<?php esc_html_e( 'Dismiss this (but don\'t enable publish tweets)', 'autoshare-for-twitter' ); ?>
+					</a>
+					<a class="button" href="<?php echo esc_url( $update_url ); ?>">
+						<?php esc_html_e( 'Enable publish tweets', 'autoshare-for-twitter' ); ?>
+					</a>
+				</p>
+			</div>
+			<?php
+		}
+	}
+
+	/**
+	 * Determines if this is a testing/staging site.
+	 *
+	 * Checks if the WordPress site URL is the same as the URL Autoshare considers the live URL.
+	 *
+	 * @since 1.2.0
+	 * @return bool Whether the site is a staging URL or not.
+	 */
+	public static function is_duplicate_site() {
```

Comment:
This function could do with number of comments, it's unclear what is been compared.

Can this use the `WP_ENVIRONMENT_TYPE` constant with the core function `wp_get_environment_type()`

Reply:
This function compares the site's current URL with the saved live URL in the DB to identify if the current env is prod or not. for `WP_ENVIRONMENT_TYPE`, I am not sure if all devs use this to env var to mark the environment. So, I am not feeling comfortable depending on this env var. what do you think?